CONTRIBUTING.md

```diff
@@ -0,0 +1,281 @@
+Messages should be understandable without looking at the code changes.
+A commit message like `FIX: fix another one` is an example of what not to do;
+the reader has to go look for context elsewhere.
+
+Standard prefixes to start the commit message with are:
+
+```text
+API: an (incompatible) API change (will be rare)
+BLD: change related to building xtgeo
+BUG: bug fix
+CI: related to CI/CD workflows
+CLN: code cleanup, maintenance commit (refactoring, typos, PEP, etc.)
+DEP: deprecate something, or remove a deprecated object
+DOC: documentation, addition, updates
+ENH: enhancement, new functionality
+FIX: fixes wrt to technical issues, e.g. wrong requirements.txt
+PERF: performance or bench-marking
+REV: revert an earlier commit
+REL: related to releasing xtgeo
+TST: addition or modification of tests
+```
+
+### Type Hints
+
+XTGeo requires the use of type annotations in all new feature
+developments, incorporating Python 3.10's enhanced syntax for type hints.
+This facilitates a more concise and readable style.
+
+### Style Guidelines
+
+- For Python versions prior to 3.10, include the following import for
+  compatibility:
+
+  ```python
+  from __future__ import annotations
+  ```
+
+- Use Python's built-in generics (e.g., `list`, `tuple`) directly. This
+  approach is preferred over importing types like `List` or `Tuple` from
+  the `typing` module.
+
+- Apply the new union type syntax using the pipe (`|`) for clarity and
+  simplicity. For example:
+
+  ```python
+  primes: list[int | float] = []
+  ```
+
+- For optional types, use `None` with the pipe (`|`) instead of `Optional`.
+  For instance:
+
+  ```python
+  maybe_primes: list[int | None] = []
+  ```
+
+Note: These guidelines align with PEP 604 and are preferred for all new code
+submissions and when updating existing code.
+
+## Pull Request Guidelines
+
+Before you submit a pull request, check that it meets these guidelines:
+
+1. The pull request should include tests.
+
+2. If the pull request adds functionality, the docs should be updated. Put
+   your new functionality into a function with a docstring and if it is a
+   public method make sure it displays helpful information in the
+   documentation.
+
+## Tips
+
+- To run a subset of tests, e.g. only surface tests:
+
+  ```python
+  pytest test/test_surfaces
+  ```
+
+- scikit-build-core offers some suggestions about building with editable
+  installs, see info here:
+
+  https://scikit-build-core.readthedocs.io/en/latest/configuration.html#editable-installs
+
+
+## Working with RMS python
+
+The following is a special recipe when working with RMS' Python version,
+and it is targeted to Equinor usage using bash shell in Linux:
+
+```sh
+unset PYTHONPATH  # to avoid potential issues
+# activate RMS python, e.g. RMS version 13.1.2
+source /prog/res/roxapi/aux/roxenvbash 13.1.2
+# make a virtual env (once):
+python -m venv ~/venv/py38_rms13.1.2
```

Comment:
Consider here to add `system-site-packages`:

`python -m venv ~/venv/py38_rms13.1.2 --system-site-packages`   # to include the shipped libraries from RMS